src/backup.cpp

```diff
@@ -93,17 +93,168 @@ bool BackupWallet(const CWallet& wallet, const std::string& strDest)
+
+    // Nothing to do if manage_backup_retention is not set, which is the default to be
+    // safe (i.e. retain backups indefinitely is the default behavior).
+    if (!manage_backup_retention) return true;
+
+    // Zeroes for both incoming retention arguments mean that the config file settings should be used.
+      if (!retention_by_num && !retention_by_days)
+    {
+        // If either argument is set, then assign the one that is set.
+        if (IsArgSet("-walletbackupretainnumfiles") || IsArgSet("-walletbackupretainnumdays"))
+        {
+            // Default to zero for the unset argument, which means unset here. Also, clamp
+            // to zero for nonsensical negative values. That kind of stupidity will be
+            // caught and dealt with below.
+            retention_by_num = (unsigned int) std::max((int64_t) 0, GetArg("-walletbackupretainnumfiles", 0));
+            retention_by_days = (unsigned int) std::max((int64_t) 0, GetArg("-walletbackupretainnumdays", 0));
+        }
+        else
+        {
+            // Default to 365 for each. (A very conservative setting.)
+            retention_by_num = (unsigned int) GetArg("-walletbackupretainnumfiles", 365);
+            retention_by_days = (unsigned int) GetArg("-walletbackupretainnumdays", 365);
+        }
+     }
+
+     // This is a conditional clamp that checks for nonsensical values to protect people.
+     // If -managebackupretention is set, but the other two are not, they both will default to
+     // 365, which will pass this check. What this does is detect a scenario where both are
+     // set to something less than 7, which is probably not a good idea. One of them can be set
+     // to less than seven, or even not set (which is the same as zero), as long as the other
+     // set to 7 or more. If both are set to less than seven, then they both will be clamped
+     // to 7. Remember the retention will follow whichever results in more files being
+     // retained, so the "and" condition is sufficient here.
+     if (retention_by_num < 7 && retention_by_days < 7)
+     {
+         LogPrintf("ERROR: ManageBackups: Nonsensical values specified for backup retention. "
+                   "Clamping both number of files and number of days to 7. Retention will follow "
+                   "whichever results in the most retention to be safe.");
+
+         retention_by_num = 7;
+         retention_by_days = 7;
+     }
+
+    LogPrintf ("INFO: ManageBackups: number of files to retain %i, number of days to retain %i. "
+               "The retention will follow whichever results in the greater number of files "
+               "retained.", retention_by_num, retention_by_days);
+
+    int64_t retention_cutoff_time = GetAdjustedTime() - retention_by_days * 86400;
+
+    // Iterate through the log archive directory and delete the oldest files beyond the retention rules.
+    // The names are in format <file type>-YYYY-MM-DDTHH-MM-SS for the backup entries, so iterate
```

Comment:
I think there is a typo in this comment?  Seems it should be `<file type>-YYYY-MM-DD-HH-MM-SS`

Reply:
No, there really is a T in there... :)